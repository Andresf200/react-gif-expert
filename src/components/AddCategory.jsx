import { useState } from "react";
import {PropTypes} from 'prop-types';

const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('')  

    const handleSubmit = (e) => {
      e.preventDefault();
     
      if(inputValue.trim() <= 1) return;
      //Podemos utilizar el callback para llamar nuestro valor anterior y remplanzar
      onNewCategory(inputValue.trim());
      setInputValue('');
    };

    const handleInputChange = (e) => {
      setInputValue(e.target.value);
    };
  
      return (
    <form onSubmit={handleSubmit} aria-label="form">
        <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={handleInputChange}
         / >
    </form>
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}

export {AddCategory}
