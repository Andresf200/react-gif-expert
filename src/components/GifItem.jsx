import PropTypes from 'prop-types';


const GifItem = ({url,title,id}) => {
  return (   
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}


GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}


export {GifItem}
