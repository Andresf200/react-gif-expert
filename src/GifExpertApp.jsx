import { useState } from "react"
import {AddCategory,GifGrid} from "./components";

function GifExpertApp() {
    const [categories, setCategories] = useState(['Goku']);

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory)) return; 
        setCategories([...categories,newCategory]); 
    };
  return (
    <>
        <h1>GifExpertApp</h1>
        <AddCategory 
            onNewCategory={onAddCategory}
        />
 
        {categories.map(category => (
                <GifGrid 
                    key={category}
                    category={category}
                />
            )
        )}
    </>
  )
}

export default GifExpertApp