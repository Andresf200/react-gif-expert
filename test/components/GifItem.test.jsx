import { GifItem } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe('Se valida el componente GifItem',() => {
    const title = "Saitama";
    const url = "http://satiama.com/sati.jpg";


    test('Validamos el snapshot', () => {       
        const {container} = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el alt y src de la imagen',() => {
       render(<GifItem title={title} url={url} />);
       //screen.debug(); // para ver como esta la pantalla osea el screenshot

       //expect(screen.getByRole('img').src).toBe(url);
       //expect(screen.getByRole('img').alt).toBe(alt);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe de mostrar el titulo del componente', () => {
       render(<GifItem title={title} url={url} />);

       expect(screen.getByText(title)).toBeTruthy();
    });
});