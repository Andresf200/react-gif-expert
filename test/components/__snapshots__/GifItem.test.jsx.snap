// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se valida el componente GifItem Validamos el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://satiama.com/sati.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
